### PR TITLE
Otp

### PR DESCRIPTION
**Pull request by Awesomeman007**
- request otp
- verify otp
- otp api swagger

### CODE REVIEW

File: src/auth/auth.module.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Module } from "@nestjs/common"
+import { AuthService } from "./auth.service"
+import { AuthController } from "./auth.controller"
+@Module({
+  imports: [],
+  providers: [AuthService],
+  controllers: [AuthController]
```
Comment (ManapatWeeramongkolkul) [inline src/auth/auth.module.ts]: Could you add an export line here? Like this: 
@Module({
  imports: [],
  providers: [AuthService],
  controllers: [AuthController],
  exports: [AuthService],
})

**Review by ManapatWeeramongkolkul — CHANGES_REQUESTED** [review]
Pls add an export line so that I can use it in mine. Everything else looks good to me.

**Review by ManapatWeeramongkolkul — APPROVED** [review]
Looks good to me

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    //       secret: process.env.OTP_SECRETKEY || 'b97cef0144f23791501a33afe87a23f5',
+    //       msisdn: phoneNumber,
+    //     },
+    //     { Accept: 'application/json' }
+    //   )
+    //   return res
+    // } catch (err) {
+    //   console.error(err)
+    //   throw new HttpException('Failed Requesting OTP', HttpStatus.BAD_REQUEST)
+    // }
```
Comment (tum210544) [inline src/auth/auth.service.ts]: are u gonna use the code that are commended again? if not please delete unused code

File: src/auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    //   )
+    //   console.log(res)
+    //   console.log('verification successful')
+    //   return { status: res.status, message: res.message }
+    // } catch (err) {
+    //   console.log('verifiacation failed')
+    //   console.log(err)
+    //   throw new HttpException('Failed Verifying OTP', HttpStatus.UNAUTHORIZED)
+    // }
+
```
Comment (tum210544) [inline src/auth/auth.service.ts]: this too

File: src/auth/dto/auth.dto.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { ApiProperty } from '@nestjs/swagger'
+import {IsString} from 'class-validator'
+// import { Type } from 'class-transformer'
```
Comment (tum210544) [inline src/auth/dto/auth.dto.ts]: this import line too

**Review by tum210544 — CHANGES_REQUESTED** [review]

**Review by tum210544 — CHANGES_REQUESTED** [review]
One big concern(but it can be fix easily): I think the name of folder should be changed (from auth --> otp) since auth is seem like authentication(jwt thing) which is not this one. Therefore, I think the name of folder along with controller, module, service, and service spec should be changed. U can change the name of the file and folder in the vs code directly and then it will ask u whether u want to change this in other file that u import or refer to the file name.

File: src/auth/auth.service.ts (hunk 2 above)
Comment (Awesomeman007) [reply src/auth/auth.service.ts]: yes this is when i want to connect to the actual third party company providing otp service

File: src/auth/auth.service.ts (hunk 3 above)
Comment (Awesomeman007) [reply src/auth/auth.service.ts]: same approach here.

**Comment by Awesomeman007** [comment]
> One big concern(but it can be fix easily): I think the name of folder should be changed (from auth --> otp) since auth is seem like authentication(jwt thing) which is not this one. Therefore, I think the name of folder along with controller, module, service, and service spec should be changed. U can change the name of the file and folder in the vs code directly and then it will ask u whether u want to change this in other file that u import or refer to the file name.

I was thinking about this issue too. But I agree to split between otp and auth.

File: src/auth/auth.service.ts (hunk 2 above)
Comment (tum210544) [reply src/auth/auth.service.ts]: oh my bad

**Review by tum210544 — APPROVED** [review]

**Pull request merged**